api/login/login.controller.js

```diff
@@ -0,0 +1,36 @@
+import UserService from '../user/user.service';
+import jwt from 'jsonwebtoken';
+import secret from '../secret';
+
+class LoginController {
+
+  login(req, res, next) {
+    const user = req.body.user;
+    const searchCondition = {name: user.username, password: user.password};
+    const token = jwt.sign(user, secret);
+
+    UserService.findBy(searchCondition, (err, foundUser) => {
+      if (err) return next(err);
+
+      //If there's no user, then we get an empty array
+      if (foundUser.length === 0) {
+        if (req.body.signUp) {
+          UserService.create({name: user.username, email: 'fake@fake.com', password: user.password}, (err, foundUser) => {
```

Comment:
Would you mind adding a `// TODO` here explaining this is temporal? Having a registration action inside a login is _weird_.